lib/Projects/ProjectGitSyncer.php

```diff
@@ -45,8 +46,8 @@ public function initRepository(string $repositoryName): void
 
         $command = sprintf(
             'git clone https://github.com/doctrine/%s.git %s',
-            $repositoryName,
-            $repositoryPath
+            escapeshellarg($repositoryName),
+            escapeshellarg($repositoryPath)
```

Comment:
I see the processFactory below uses `symfony/process`, which has support for
[escaping arguments](https://github.com/symfony/process/blob/240e74140d4d956265048f3025c0aecbbc302d54/Process.php#L177-L178):

```
     *   $process = Process::fromShellCommandline('my_command "$MY_VAR"');
     *   $process->run(null, ['MY_VAR' => $theValue]);
```

That would imply adding an extra argument for processFactory som that this can
be changed to

```php
$this->processFactory->run(
    'git clone "https://github.com/doctrine/$REPOSITORY_NAME.git" $REPOSITORY_PATH',
    null,
    ['REPOSITORY_NAME' => $repositoryName, 'REPOSITORY_PATH' => $repositoryPath]
);`
```

Reply:
It looks like unnecessary coupling to the Process API to mo.

Reply:
I think it's fair to say anything that deals with processes will have an associative array at some point, just like any library that deals with prepared SQL statements will have such arrays for binding named parameters. And the `ProcessFactory` abstraction is here to help making sure we can still ditch `symfony/process` and use something else while keeping that API, isn't it?

Reply:
In this code, you just need to build a shell command which later will be executed. Building an abstraction over another abstraction looks like overkill and premature generalization.

Reply:
Is it that premature though? There are already many calls to `run` combined with `sprintf` in the codebase, and that PR adds more of them. Looks more overdue than premature to me.

https://github.com/doctrine/doctrine-website/search?q=this-%3EprocessFactory-%3Erun

You will note that existing calls neglect to do the escaping, and rely on a `$command` OTV most of the times. Moreover, we already have an abstraction (`ProcessFactory` built over `symfony/process`).

Reply:
@greg0ire do you want to discuss this as part of this PR?

Reply:
I don't require we do that change as part of this PR, but it's still a good occasion to discuss it. Feel free to merge as is, I don't want this to be a blocker, and I might do the refactoring myself in a subsequent, so that the benefits are easier to spot.

Reply:
It's easy to go off track with a website PR. There have to be a few more changes, but one small step at the time.

Thanks for the reviews.